### PR TITLE
Precompiled Message Log and Preview letter tests

### PR DESCRIPTION
**Pull request by richardc0**
**Message Log**

Added extra assertion to the precompiled letter tests to check that the link in the message log is enabled when the letter has been successfully virus scanned. If the link is present and matches the predicted link it checks that the page can be visited and that there is a precompiled
letter text on the page. Further checks to this page will be added in a new test navigated from the main admin page rather than the message log.

* Added extra check to existing delivered and failed tests
* Added extra method to the page utils class
* Created a new class for the preview letter page
* Added a new locator to find the link on the message log page

**Letter Preview**

Added tests to check the letter preview page for a precompiled letter this checks the text on that page, checks the link to the PDF and also the image isn't the static error image.

* Refactored message log test to split out message log and preview functionality
* Added extra asserts to check for the pdf link and the image
* Added locators for the letter preview page
* Added new methods to the PreviewLetterPage class

### CODE REVIEW

File: tests/functional/preview_and_dev/test_seeded_user.py [hunk 1]
```diff
@@ -182,6 +184,68 @@ def test_view_precompiled_letter_message_log_delivered(
         delay=Config.NOTIFICATION_RETRY_INTERVAL
     )
 
+    ref_link = "https://www.notify.works/services/" + \
```
Comment (servingUpAces) [inline tests/functional/preview_and_dev/test_seeded_user.py]: Should the host be hard coded here?

Comment (servingUpAces) [inline tests/functional/preview_and_dev/test_seeded_user.py]: hard coded host name?

File: tests/functional/preview_and_dev/test_seeded_user.py [hunk 2]
```diff
@@ -207,6 +271,13 @@ def test_view_precompiled_letter_message_log_virus_scan_failed(
...
```
Comment (servingUpAces) [inline tests/functional/preview_and_dev/test_seeded_user.py]: hard coded host name?

File: tests/functional/preview_and_dev/test_seeded_user.py (hunk 1 above)
Comment (richardc0) [reply tests/functional/preview_and_dev/test_seeded_user.py]: Agreed removed the need for a host

Comment (richardc0) [reply tests/functional/preview_and_dev/test_seeded_user.py]: Agreed removed the need for a host

File: tests/functional/preview_and_dev/test_seeded_user.py [hunk 3]
```diff
@@ -182,6 +184,68 @@ def test_view_precompiled_letter_message_log_delivered(
...
+
+    api_integration_page.go_to_preview_letter()
+
+    letter_preview_page = PreviewLetterPage(driver)
+    assert letter_preview_page.is_text_present_on_page("Provided as PDF, sent on")
+
+    # Check the pdf link looks valid
+    pdf_download_link = letter_preview_page.get_download_pdf_link()
+
+    link = "https://www.notify.works/services/" + \
```
[the remark this answers is not in the document]
Comment (richardc0) [reply tests/functional/preview_and_dev/test_seeded_user.py]: Agreed removed the need for a host

File: tests/functional/preview_and_dev/test_seeded_user.py (hunk 2 above)
Comment (richardc0) [reply tests/functional/preview_and_dev/test_seeded_user.py]: Agreed removed the need for a host

**Pull request merged**